scarlet2/initialization.py

```diff
@@ -0,0 +1,331 @@
+    fn = lambda x: jnp.mean(
+        central_gaussian * (data_cut - create_gaussian_array(bx, x[0], x[1], x[2])) ** 2
+    )
+    x0 = jnp.array([1.5, 1.5, 0])
+    lbfgsb = ScipyBoundedMinimize(fun=fn, method="l-bfgs-b", jit=True)
+    lower_bounds = jnp.array([1, 1, jnp.pi / 2])
+    upper_bounds = jnp.array([7, 7, jnp.pi])
+    bounds = (lower_bounds, upper_bounds)
+    best_params = lbfgsb.run(x0, bounds=bounds).params
+    return best_params
+
+
+# take the 3 moments of a @D gaussian and construct the covariance matrix to fit the shape
+
+
+def init_simple_morph(
+    observation, center, psf_sigma=0.5, noise_thresh=20, corr_thresh=0.8
+):
+    """
+    Initialize the morphology of a source by fitting a 2D Gaussian to the cutout of the source.
+    The boxsize will initially be fit as a compact point-source and then expanded until the snr
+    is below a desired threshold.
+
+    Parameters
+    ----------
+    observation: `~scarlet.Observation`
+    center: tuple --> center of source
+    psf_sigma: float --> initial psf used in model
+    noise_thresh: float --> threshold for noise
+    corr_thresh: float --> threshold for spectrum correlations
+
+    Returns
+    -------
+    morph: `jnp.array` --> 2D Gaussian morphology
+    """
+    sigma = psf_sigma
+    converged = False
+    peak_spec = init_spectrum(observation, center, correct_psf=True)
+    spectra_prev = peak_spec
+    noise_rms = 1 / np.sqrt(ma.masked_equal(observation.weights, 0))
+    ma.set_fill_value(noise_rms, np.inf)
+
+    # loop over box size until SNR is below threshold
+    while converged == False:
+        morph = GaussianMorphology(center, sigma=sigma)
+        box = morph.bbox.shape
+
+        # now grab the perimeter values of the box
+        perimeter_pixels = [None] * len(observation.data)
+        perimeter_noise_rms = [None] * len(observation.data)
+
+        # create the cutout and noise rms for each band
+        for idx, band in enumerate(observation.data):
+            cutout_obs = cut_square_box(band, center, box[0])
+            cutout_noise = cut_square_box(noise_rms[idx], center, box[0])
+            perimeter_pixels[idx] = jnp.array([perimeter_values(cutout_obs)])
+            perimeter_noise_rms[idx] = jnp.array([perimeter_values(cutout_noise)])
+        perimeter_pixels = jnp.squeeze(jnp.stack(perimeter_pixels, axis=0), axis=1)
+        perimeter_noise_rms = jnp.squeeze(
+            jnp.stack(perimeter_noise_rms, axis=0), axis=1
+        )
+
+        # simplified spectrum calculation for perimeter average around cutout
+        models = (None,) * len(observation.data)
+        spectra = []
+        spectrum_avg = 0
+        for obs, model in zip((observation,), models):
+            for i in range(perimeter_pixels.shape[1]):
+                spectrum = perimeter_pixels[:, i]
+                psf_model = obs.frame.psf()
+                # psf_model = jnp.expand_dims(psf_model, axis=0)
+                psf_peak = psf_model.max(axis=(1, 2))
```

Comment:
That should be outside of the `for i ... ` loop

Reply:
Fixed